optimism/src/keccak/column.rs

```diff
@@ -274,3 +278,148 @@ impl<T: Clone> IndexMut<KeccakColumn> for KeccakColumns<T> {
         }
     }
 }
+
+impl<G> IntoParallelIterator for KeccakColumns<G>
+where
+    Vec<G>: IntoParallelIterator,
+{
+    type Iter = <Vec<G> as IntoParallelIterator>::Iter;
+    type Item = <Vec<G> as IntoParallelIterator>::Item;
+
+    fn into_par_iter(self) -> Self::Iter {
+        let mut iter_contents = Vec::with_capacity(ZKVM_KECCAK_COLS_LENGTH);
+        iter_contents.push(self.hash_index);
+        iter_contents.push(self.step_index);
+        iter_contents.push(self.flag_round);
+        iter_contents.push(self.flag_absorb);
+        iter_contents.push(self.flag_squeeze);
+        iter_contents.push(self.flag_root);
+        iter_contents.push(self.flag_pad);
+        iter_contents.push(self.flag_length);
+        iter_contents.push(self.two_to_pad);
+        iter_contents.push(self.inverse_round);
+        iter_contents.extend(self.flags_bytes);
+        iter_contents.extend(self.pad_suffix);
+        iter_contents.extend(self.round_constants);
+        iter_contents.extend(self.curr);
+        iter_contents.extend(self.next);
+        iter_contents.into_par_iter()
+    }
+}
+
+impl<G: Send + std::fmt::Debug> FromParallelIterator<G> for KeccakColumns<G> {
+    fn from_par_iter<I>(par_iter: I) -> Self
+    where
+        I: IntoParallelIterator<Item = G>,
+    {
+        let mut iter_contents = par_iter.into_par_iter().collect::<Vec<_>>();
+        let next = iter_contents
+            .drain(iter_contents.len() - ZKVM_KECCAK_COLS_NEXT..)
```

Comment:
I would guess it is fine as the length is modified. We'll see later with the final code. Don't know if there is a better way to achieve this.

Reply:
Yeah, that was my impression, `drain()` acts as a repeated `pop()`. But in any case, we will find out in the small test I want to add to this file.